### PR TITLE
[Doc] add supported data types and delete obsolete desc in show partitions

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: docs/sql-reference/sql-statements/data-manipulation/SHOW PARTITIONS.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ SHOW [TEMPORARY] PARTITIONS FROM [db_name.]table_name [WHERE] [ORDER BY] [LIMIT]
 
 > NOTE
 >
-> This syntax only supports StarRocks tables (`"ENGINE" = "OLAP"`). For Elasticsearch and Hive tables, use SHOW PROC '/dbs/db_id/table_id/partitions'.
+> This syntax only supports StarRocks tables (`"ENGINE" = "OLAP"`).
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-manipulation/SHOW PARTITIONS.md:15]: Suggestion: This syntax only supports StarRocks tables (namely, tables for which `"ENGINE" = "OLAP"` is specified at table creation).

File: docs/table_design/Data_distribution.md [hunk 2]
```diff
@@ -469,6 +469,7 @@ If partition data cannot be evenly distributed across all the buckets by using o
 #### Precautions
 
 - **When a table is created, you must specify the bucketing columns**.
+- The data types of bucketing columns are supported to be INTEGER, DECIMAL, DATE/DATETIME, CHAR/VARCHAR/STRING.
```
Comment (amber-create) [inline docs/table_design/Data_distribution.md:472]: Suggestion: The data types of bucketing columns must be INTEGER, DECIMAL, DATE/DATETIME, or CHAR/VARCHAR/STRING.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.1 branch-3.0 branch-2.5

**Pull request merged**